File: apps/client/backoffice/src/app/utils/status.enum.ts
export enum StatusEnum {
	LIVE = 'LIVE',
	TRACKED = 'TRACKED',
	PAID = 'PAID',
	FINISHED = 'FINISHED',
	STOPPED = 'STOPPED',
	REJECTED = 'REJECTED',
	VALIDATED = 'VALIDATED',
	EXPIRED = 'EXPIRED',
	PENDING = 'PENDING',
	COMPLETED = 'COMPLETED',
	CONFIRMED = 'CONFIRMED',
	CANCELLED = 'CANCELLED',
	REQUESTED = 'REQUESTED',
}

export const transactionStatuses: StatusEnum[] = [
	StatusEnum.TRACKED,
	StatusEnum.VALIDATED,
	StatusEnum.REJECTED,
];

export const rewardStatuses: StatusEnum[] = [
	StatusEnum.TRACKED,
	StatusEnum.CONFIRMED,
	StatusEnum.REJECTED,
	StatusEnum.LIVE,
	StatusEnum.EXPIRED,
	StatusEnum.STOPPED,
	StatusEnum.PAID,
	StatusEnum.FINISHED,
];

export enum CashbackTypeEnum {
	FIXED = 'FIXED',
	PERCENTAGE = 'PERCENTAGE',
}

export enum SortListEnum {
	LATEST_STORES = 'created_at desc',
	BY_TITLE = 'name',
	MOST_POPULAR = 'most-popular',
}

File: apps/client/mobile/lib/utils/enums/status_enum.dart
enum StatusEnum {
  live('LIVE'),
  tracked('TRACKED'),
  paid('PAID'),
  finished('FINISHED'),
  stopped('STOPPED'),
  rejected('REJECTED'),
  validated('VALIDATED'),
  expired('EXPIRED'),
  pending('PENDING'),
  completed('COMPLETED'),
  confirmed('CONFIRMED'),
  cancelled('CANCELLED'),
  requested('REQUESTED'),

  active('ACTIVE'),

  inactive('INACTIVE');

  const StatusEnum(this.value);

  final String value;

  static StatusEnum fromString(String value) {
    for (StatusEnum status in StatusEnum.values) {
      if (status.value == value) {
        return status;
      }
    }
    return StatusEnum.live;
  }

  @override
  String toString() => value;
}
